uaclient/files/state_files.py

```diff
@@ -0,0 +1,49 @@
+from typing import Any, Dict
+
+from uaclient.data_types import BoolDataValue, DataObject, Field
+from uaclient.files.data_types import DataObjectFile
+from uaclient.files.files import UAFile
+
+SERVICES_ONCE_ENABLED = "services-once-enabled"
+
+
+class _ServicesOnceEnabledData(DataObject):
+    fields = [
+        Field("fips_updates", BoolDataValue, False),
+    ]
+
+    def __init__(self, fips_updates: bool):
+        self.fips_updates = fips_updates
+
+
+class _ServicesOnceEnabledDataObjectFile(
+    DataObjectFile[_ServicesOnceEnabledData]
+):
+    def add_service(self, service: str):
+        file_content = self.read()
+        services_dict = {} if not file_content else file_content.to_dict()
+        service = service.replace("-", "_")
+
+        if service not in services_dict:
+            services_dict[service] = True
+            self.write(_ServicesOnceEnabledData(**services_dict))
```

Comment:
Sorry, I think I am missing the point here. Why should we only support `fips-updates` here ? Even though it is the only service using that logic, we should still make it generic if we need it for other services

Reply:
The function as currently written only supports fips-updates and will already raise an exception if any other service name is passed. That is because `_ServicesOnceEnabledData` only has a field for `fips_updates`, so `_ServicesOnceEnabledData(**services_dict)` will raise an exception if any other service is passed.

We could change it to `_ServicesOnceEnabledData.from_dict(services_dict)`, but then it still wouldn't work for other services. it would just silently swallow the other services and not work, which we might not catch in the future.

So my suggestion is that we do one of the following:
* Option A: support all services by adding the necessary fields to `_ServicesOnceEnabledData`
* Option B: make it more explicit that this only supports fips-updates currently (that is what my previous comment in this thread is trying to do)
* Option C: don't add the `add_service` method, and instead directly use write and read wherever needed

(Option C is what I was advocating for in a different thread but really any of them are okay IMO)

Reply:
Ohhh great catch. Fair enough, thinking better about this method I think you are right regarding option C. If in the future we want to support more services we can add back that method